File: server/core/service/user_service.go
package service

import (
	"errors"

	"github.com/nexus-flow/nexus/core/repository"
	"github.com/nexus-flow/nexus/pkg/auth"
)

type LoginResult struct {
	UserName string `json:"userName"`
	Token    string `json:"token"`
}

type UserInfo struct {
	ID       int64  `json:"id"`
	UserName string `json:"userName"`
}

func Login(userName, password string) (*LoginResult, error) {
	user, err := repository.GetUserByName(userName)
	if err != nil {
		return nil, errors.New("用户名或密码错误")
	}
	if password != user.Password {
		return nil, errors.New("用户名或密码错误")
	}
	token, err := auth.GenerateToken(user.ID)
	if err != nil {
		return nil, errors.New("生成Token失败")
	}
	return &LoginResult{
		UserName: user.UserName,
		Token:    token,
	}, nil
}

func GetUserInfo(userID int64) (*UserInfo, error) {
	user, err := repository.GetUserByID(userID)
	if err != nil {
		return nil, errors.New("用户不存在")
	}
	return &UserInfo{
		ID:       user.ID,
		UserName: user.UserName,
	}, nil
}

func UpdatePassword(userID int64, oldPwd, newPwd string) error {
	user, err := repository.GetUserByID(userID)
	if err != nil {
		return errors.New("用户不存在")
	}
	if oldPwd != user.Password {
		return errors.New("原密码错误")
	}
	return repository.UpdateUserPassword(userID, newPwd)
}

File: server/src/main/java/com/javaoffers/nexus/core/service/UserService.java
package com.javaoffers.nexus.core.service;

import com.javaoffers.nexus.core.model.User;
import com.javaoffers.nexus.core.repository.UserMapper;
import com.javaoffers.nexus.pkg.auth.JwtUtil;
import com.javaoffers.nexus.pkg.exception.BizException;
import lombok.Data;
import org.springframework.stereotype.Service;

/** 对应 Go core/service/user_service.go。 */
@Service
public class UserService {

    private final UserMapper userMapper;
    private final JwtUtil jwtUtil;

    public UserService(UserMapper userMapper, JwtUtil jwtUtil) {
        this.userMapper = userMapper;
        this.jwtUtil = jwtUtil;
    }

    @Data
    public static class LoginResult {
        private String userName;
        private String token;
    }

    @Data
    public static class UserInfo {
        private Long id;
        private String userName;
    }

    public LoginResult login(String userName, String password) {
        User user = userMapper.getByName(userName);
        if (user == null) {
            throw new BizException("用户名或密码错误");
        }
        if (password == null || !password.equals(user.getPassword())) {
            throw new BizException("用户名或密码错误");
        }
        LoginResult r = new LoginResult();
        r.setUserName(user.getUserName());
        r.setToken(jwtUtil.generateToken(user.getId()));
        return r;
    }

    public UserInfo getUserInfo(Long userId) {
        User user = userMapper.getById(userId);
        if (user == null) {
            throw new BizException("用户不存在");
        }
        UserInfo info = new UserInfo();
        info.setId(user.getId());
        info.setUserName(user.getUserName());
        return info;
    }

    public void updatePassword(Long userId, String oldPwd, String newPwd) {
        User user = userMapper.getById(userId);
        if (user == null) {
            throw new BizException("用户不存在");
        }
        if (oldPwd == null || !oldPwd.equals(user.getPassword())) {
            throw new BizException("原密码错误");
        }
        userMapper.updatePassword(userId, newPwd);
    }
}
